analysis/Analysis/Section_3_6.lean

```diff
@@ -290,14 +290,112 @@ theorem SetTheory.Set.two_to_two_iff {X Y:Set} (f: X → Y): Function.Injective
 
 /-- Exercise 3.6.12 -/
 def SetTheory.Set.Permutations (n: ℕ): Set := (Fin n ^ Fin n).specify (fun F ↦
-    Function.Bijective ((powerset_axiom F).mp F.prop).choose)
+    Function.Bijective (pow_fun_equiv F))
 
-/-- Exercise 3.6.12 (i) -/
+/-- Exercise 3.6.12 (i), first part -/
 theorem SetTheory.Set.Permutations_finite (n: ℕ): (Permutations n).finite := by sorry
 
-/-- Exercise 3.6.12 (i) -/
+/- To continue Exercise 3.6.12 (i), we'll first develop some theory about `Permutations` and `Fin`. -/
+
+noncomputable def SetTheory.Set.Permutations_toFun {n: ℕ} (p: Permutations n) : (Fin n) → (Fin n) := by
```

Comment:
There's probably some way to write this without `by` but I got lost trying to extract `mem_Permutations` and gave up on that.